Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -164,6 +164,9 @@ class SyscallHandler : public FEXCore::HLE::SyscallHandler {
 
   FEX::HLE::MemAllocator *Get32BitAllocator() { return Alloc32Handler.get(); }
 
+  virtual void *GuestMmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset) = 0;
```

Comment:
Oh yea that'd be totally doable. Will implement.

Reply:
Looked into this today, and I'd rather keep the overridable functions. They are implemented where the rest of the memory related syscalls are, and it's easy to figure out that the tracking is there -- otherwise it's kind of weird these two are missing from there.

Reply:
As mentioned I'm okay with keeping it non-virtual, but this really needs a docstring.

Reply:
I've added some basic docstrings, though in my view what these functions do is pretty self explanatory from their name.

They don't have to call Track* functions, they have to do a guest mmap, which with the current way the mman syscalls work have to do the tracking, so it's not some special requirement.